### PR TITLE
Update DELETE SQL Diagram

### LINKED ISSUE #2258: UPDATE and DELETE now support ORDER BY and LIMIT clauses
PR: https://github.com/cockroachdb/cockroach/pull/20069

- For [`UPDATE`]()https://www.cockroachlabs.com/docs/dev/update.html), this is a MySQL extension that can help with updating the primary key of a table (`ORDER BY`) and control the maximum size of write transactions (`LIMIT`).
- For [`DELETE`](https://www.cockroachlabs.com/docs/dev/delete.html), the `ORDER BY` clause constrains the deletion order, the output of its `LIMIT` clause (if any), and the result order of its `RETURNING` clause (if any).

### PR DESCRIPTION
**Pull request by lhirata**
Closes #2258.

`UPDATE` edits moved to #2514.

### CODE REVIEW

File: v2.0/delete.md [hunk 1]
```diff
@@ -31,10 +31,12 @@ table td:first-child {
 | Parameter | Description |
 |-----------|-------------|
 |`relation_expr` | The name of the table you want to delete rows from.<br><br>Deleting from multiple tables in a single statement is not supported.|
-|`AS name` | Create an alias for the table name, completely hiding its original name. All subsequent references to the table must use its alias.<br><br>Aliases are primarily used with `JOIN`, which is not yet supported but is coming in a [future release](https://github.com/cockroachdb/cockroach/issues/2970).|
+|`AS table_alias_name` | Create an alias for the table name, completely hiding its original name. All subsequent references to the table must use its alias.<br><br>Aliases are primarily used with `JOIN`, which is not yet supported but is coming in a [future release](https://github.com/cockroachdb/cockroach/issues/2970).|
```
Comment (jseldess) [inline v2.0/delete.md]: Please remove that second paragraph about JOIN not yet being supported. That's way out of date. Can you also remove it from wherever you copied this text as well?

File: v2.0/delete.md [hunk 2]
```diff
@@ -31,10 +31,12 @@ table td:first-child {
...
 |`WHERE a_expr`| `a_expr` must be an expression that returns Boolean values using columns (e.g. `<column> = <value>`). Delete rows that return `TRUE`.<br><br/>__Without a `WHERE` clause in your statement, `DELETE` removes all rows from the table.__|
+| `ORDER BY sortby_list` | Sort retrieved rows in the order of comma-separated column names you include in `sortby_list`. You can optionally specify `ASC` or `DESC` order for each column.<br><br>When <code>ORDER BY</code> is not included in a query, rows are not sorted by any consistent criteria. Instead, CockroachDB returns them as the coordinating node receives them.|
+| `LIMIT count` | Only retrieve `count` number of rows. <br><br>CockroachDB also supports `FETCH FIRST count ROWS ONLY` and `FETCH NEXT count ROWS ONLY` as aliases for `LIMIT`. If `count` is omitted, then one row is fetched.|
```
Comment (jseldess) [inline v2.0/delete.md]: Let's say `For compatibility with PostgreSQL, CockroachDB also supports...`

Comment (lhirata) [reply v2.0/delete.md]: Done.

File: v2.0/delete.md (hunk 1 above)
Comment (lhirata) [reply v2.0/delete.md]: Updated. This was only written here.

**Pull request merged**